extensions/amp-form/0.1/amp-form.js

```diff
@@ -556,14 +556,14 @@ export class AmpForm {
       // Validity checking should always occur, novalidate only circumvent
       // reporting and blocking submission on non-valid forms.
       const isValid = checkUserValidityOnSubmission(this.form_);
-      if (this.shouldValidate_ && !isValid) {
+      if (this.shouldValidate_) {
```

Comment:
I think it only reports when the form is invalid:

> ...returns true if the element's child controls satisfy their validation constraints. When false is returned, cancelable invalid events are fired for each invalid child and validation problems are reported to the user.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reportValidity